### DISCUSSION TITLE
Otel-collector not able to accept rabbitmq-prometheus plugin metrics

### DISCUSSION THREAD

**Discussion by amolngt** [open-telemetry/opentelemetry-collector#11767] [Q&A]
Hi all, 

I am working on apache skywalking with OAP, otel collector, banyandb. I want  rabbitmq-prometheus metrics (http://localhost:15692/metrics/)  should scraped and export by the OTel Collector. At the end i want these metrics for creating alarm rules in apache skywalking.
there is a metrics called rabbitmq_queue_messages_unacked coming from rabbitmq-prometheus . But otel-collector is not able to scrape it. I am getting below error in OAP docker container->

java.lang.IllegalArgumentException: org.apache.skywalking.mqe.rt.exception.IllegalExpressionException: Expression: sum(rabbitmq_queue_messages_unacked) > 100 error: Metric: [rabbitmq_queue_messages_unacked] does not exist.

It would be really appreciating if you provide me some example or doc.

otel-collector-config.yaml file

```
receivers:
  prometheus:
    config:
      scrape_configs:
        - job_name: 'rabbitmq-monitoring'
          metrics_path: "/metrics"
          scrape_interval: 5s
          static_configs:
            - targets: ['ip:15692']

processors:
  batch:

exporters:
  otlp:
    endpoint: ip:11800
    tls:
      insecure: true

service:
  pipelines:
    metrics:
      receivers: [prometheus]
      processors: [batch]
      exporters: [otlp]
```

Let me know if you need more details.
Thank you in advance.

**Comment by amolngt** [comment]
Hi all,
I am still working on same issue. i understood few things. rabbitmq-prometheus plugin is able to generate metrics like rabbitmq_queue_messages_unacked by addding below config in rabbitmq.conf file

```
prometheus.tcp.port = 15692
prometheus.return_per_object_metrics = true
```
but prometheus which is getting used inside skywalk- OAP (running on port 9090) is not able to scrape it. if i use separate prometheus on different port then it work, but OAP internal prometheus doesn't allow that.
basically i want rabbitmq_queue_messages_unacked (queue wise messages count) metric which i can use in OAP alarm-settings to create a rule and send an alarm. but i get error(from OAP) ->

`java.lang.IllegalArgumentException: org.apache.skywalking.mqe.rt.exception.IllegalExpressionException: Expression: rabbitmq_queue_messages_unacked > 100 error: Metric: [rabbitmq_queue_messages_unacked] does not exist.`

Thank you in advance.
